### ISSUE TITLE
relative uri 

### ISSUE THREAD

**Issue by dpastoor** [typesense/typesense-js#10]
Hi,

Is there any way to use the typesense client in a way that does not hard code the entire protocol? Namely, I would like to use relative url (eg `/typesense`) such that if the url changes the interaction would still work.

As of now I guess the solution would be to manually extract that information from the `window.location`, however that does not feel like a great option.

Any thoughts?

**Comment by kishorenc** [comment]
In order to keep the clients uniform, we will probably not be able to do this. But, here's a easy way to get the host: instead of parsing`window.location` properties individually and stringing them together, you can simply use `window.location.origin` to get the fully qualified host name in a format like this: 

`https://github.com`

**Comment by dpastoor** [comment]
Yes, I understand these things are _possible_, its just not really idiomatic for javascript clients

For example the documentation for axios (and just about every other api example for any client library) shows client examples like so:

```
// Send a POST request
axios({
  method: 'post',
  url: '/user/12345',
  data: {
    firstName: 'Fred',
    lastName: 'Flintstone'
  }
});
```

whereas we now need to

```
`${window.location.origin}/${apiPrefix}`
```

**Comment by mkbrechtel** [comment]
I would like to run the frontend for the typesense server on the same host as the server. It would be nice if it would be possible to not set the host in the config or have an option to let the client use a relative path. Could you please open the ticket again, this is something that is usually possible with most API clients.
